Sources/BraveWallet/Crypto/NetworkSelectionView.swift

```diff
@@ -149,8 +152,14 @@ struct NetworkSelectionView: View {
   
   private func selectNetwork(_ presentation: NetworkPresentation.Network) {
     Task { @MainActor in
-      if await store.selectNetwork(presentation) {
+      if case .formSelection = store.mode {
+        guard case let .network(network) = presentation else { return }
+        store.networkSelectionInForm = network
         presentationMode.dismiss()
+      } else {
+        if await store.selectNetwork(presentation) {
+          presentationMode.dismiss()
+        }
       }
```

Comment:
I think this logic belongs in `NetworkSelectionStore` within the `selectNetwork` function, similar to the other two modes? Then it can be tested against in `NeworkSelectionStoreTests`

Reply:
addressed in [15f37fc](https://github.com/brave/brave-ios/pull/6346/commits/15f37fc85e89719dba2dd37a433a845c1ba47f8f)